src/init.cpp

```diff
@@ -1235,6 +1235,145 @@ bool AppInitLockDataDirectory()
     return true;
 }
 
+static bool AppInitLoadBlockIndex(const CChainParams& chainparams, int64_t nCoinDBCache, int64_t nBlockTreeDBCache, bool fReindexChainState, bool& fReset, std::string& strLoadError)
+{
+    const int64_t load_block_index_start_time = GetTimeMillis();
+    bool is_coinsview_empty;
+    try {
+        LOCK(cs_main);
+        UnloadBlockIndex();
+        pcoinsTip.reset();
+        pcoinsdbview.reset();
+        pcoinscatcher.reset();
+        // new CBlockTreeDB tries to delete the existing file, which
+        // fails if it's still open from the previous loop. Close it first:
+        pblocktree.reset();
+        pblocktree.reset(new CBlockTreeDB(nBlockTreeDBCache, false, fReset));
+
+        if (fReset) {
+            pblocktree->WriteReindexing(true);
+            //If we're reindexing in prune mode, wipe away unusable block files and all undo data files
+            if (fPruneMode)
+                CleanupBlockRevFiles();
+        }
+
+        if (ShutdownRequested()) return false;
+
+        // LoadBlockIndex will load fHavePruned if we've ever removed a
+        // block file from disk.
+        // Note that it also sets fReindex based on the disk flag!
+        // From here on out fReindex and fReset mean something different!
+        if (!LoadBlockIndex(chainparams)) {
+            strLoadError = _("Error loading block database");
+            return false;
+        }
+
+        // If the loaded chain has a wrong genesis, bail out immediately
+        // (we're likely using a testnet datadir, or the other way around).
+        if (!mapBlockIndex.empty() && !LookupBlockIndex(chainparams.GetConsensus().hashGenesisBlock)) {
+            fReset = true; // don't retry
```

Comment:
was this added? I don't see it in the original block. not saying it's wrong, but just wondering.

Reply:
It's that way because there were 2 ways to exit the prior codeblock: `break` and `return`. This is special handling for the one `return` case.